app/controllers/notes_controller.rb

```diff
@@ -1,12 +1,17 @@
 class NotesController < ApplicationController
-  before_action :authorize_for_ta_and_admin
-  before_action :ensure_can_modify, only: [:edit, :update]
-
+  before_action do |_|
+    if params[:id].nil?
+      authorize! with: NotePolicy
```

Comment:
Can you and Misha check if you and just do `authorize!` without the `with: NotePolicy` part (I think this should be looked up automatically).